node/config.go

```diff
@@ -394,48 +393,23 @@ func (c *Config) NodeKey() *ecdsa.PrivateKey {
 	return key
 }
 
-// StaticNodes returns a list of node enode URLs configured as static nodes.
-func (c *Config) StaticNodes() []*enode.Node {
-	return c.parsePersistentNodes(&c.staticNodesWarning, c.ResolvePath(datadirStaticNodes))
+// CheckLegacyFiles inspects the datadir for signs of legacy static-nodes
+// and trusted-nodes files. If they exist it raises an error.
+func (c *Config) checkLegacyFiles() {
+	c.checkLegacyFile(&c.staticNodesWarning, c.ResolvePath(datadirStaticNodes))
+	c.checkLegacyFile(&c.trustedNodesWarning, c.ResolvePath(datadirTrustedNodes))
 }
 
-// TrustedNodes returns a list of node enode URLs configured as trusted nodes.
-func (c *Config) TrustedNodes() []*enode.Node {
-	return c.parsePersistentNodes(&c.trustedNodesWarning, c.ResolvePath(datadirTrustedNodes))
-}
-
-// parsePersistentNodes parses a list of discovery node URLs loaded from a .json
-// file from within the data directory.
-func (c *Config) parsePersistentNodes(w *bool, path string) []*enode.Node {
+// checkLegacyFile will only raise an error if a file at the given path exists.
+func (c *Config) checkLegacyFile(w *bool, path string) {
 	// Short circuit if no node config is present
 	if c.DataDir == "" {
-		return nil
+		return
 	}
 	if _, err := os.Stat(path); err != nil {
-		return nil
-	}
-	c.warnOnce(w, "Found deprecated node list file %s, please use the TOML config file instead.", path)
-
-	// Load the nodes from the config file.
-	var nodelist []string
-	if err := common.LoadJSON(path, &nodelist); err != nil {
-		log.Error(fmt.Sprintf("Can't load node list file: %v", err))
-		return nil
-	}
-	// Interpret the list as a discovery node array
-	var nodes []*enode.Node
-	for _, url := range nodelist {
-		if url == "" {
-			continue
-		}
-		node, err := enode.Parse(enode.ValidSchemes, url)
-		if err != nil {
-			log.Error(fmt.Sprintf("Node URL %s: %v\n", url, err))
-			continue
-		}
-		nodes = append(nodes, node)
+		return
 	}
-	return nodes
+	c.warnOnce(w, true, "Ignoring deprecated node list file %s. Please use the TOML config file instead.", path)
```

Comment:
Actually, what's the point of using `warnOnce`? 
- It uses the `config.Logger`, which is: `// Logger is a custom logger to use with the p2p.Server` .
- It uses the `w` to decide if it has already warned, in your case `&c.staticNodesWarning`. But is there practically any actual risk of that happening multiple times? 

 ```suggestion
	log.Error("Ignoring deprecated node list. Please use the TOML config file instead.", "path", path)
```

Reply:
The checkLegacyFiles is only called from here: 
```
	if node.server.Config.StaticNodes == nil || node.server.Config.TrustedNodes == nil {
		node.config.checkLegacyFiles()
	}

```
 So I don't see how that could happen multiple times, so you can just skip that whole dance with `&c.staticNodesWarning`. I think.

Reply:
Using `c.Logger` is good style. All p2p / node code uses the logger passed via config. It's useful in unit tests.